jupyterhub/templates/image-puller/_daemonset-helper.yaml

```diff
@@ -68,6 +68,15 @@ spec:
             - -c
             - echo "Pulling complete"
         {{- end }}
+        {{- range $k, $container := .Values.singleuser.extraContainers }}
+        - name: image-pull-singleuser-extra-container-{{ $k }}
+          image: {{ $container.image }}
```

Comment:
Ah yes! This matches the kubernetes container model as it should, thanks for the helpful link in another commit!